tracardi/process_engine/action/v1/connectors/elastic_email/add_contact/plugin.py

```diff
@@ -0,0 +1,185 @@
+from datetime import datetime
+from multiprocessing.pool import ApplyResult
+
+import ElasticEmail
+from ElasticEmail.api import contacts_api
+from ElasticEmail.model.contact_payload import ContactPayload
+from ElasticEmail.model.contact_status import ContactStatus
+from ElasticEmail.model.email_send import EmailSend
+from asyncio import sleep
+
+from tracardi.service.notation.dict_traverser import DictTraverser
+from tracardi.service.plugin.domain.register import Plugin, Spec, MetaData, Documentation, PortDoc, Form, FormGroup, \
+    FormField, FormComponent
+from tracardi.service.plugin.domain.result import Result
+from tracardi.service.plugin.runner import ActionRunner
+from .model.config import Config, Connection
+from tracardi.service.storage.driver import storage
+from tracardi.domain.resource import Resource
+
+
+def validate(config: dict) -> Config:
+    return Config(**config)
+
+
+class ElasticEmailContactAdder(ActionRunner):
+    @staticmethod
+    async def build(**kwargs) -> 'ElasticEmailContactAdder':
+        config = Config(**kwargs)
+        resource = await storage.driver.resource.load(config.source.id)
+        return ElasticEmailContactAdder(config, resource)
+
+    def __init__(self, config: Config, resource: Resource):
+        self.config = config
+        self.credentials = resource.credentials.get_credentials(self, output=Connection)  # type: Connection
+
+    @staticmethod
+    def parse_mapping(mapping):
+        for key, value in mapping.items():
+
+            if isinstance(value, list):
+                if key == "tags":
+                    mapping[key] = ",".join(value)
+
+                else:
+                    mapping[key] = "|".join(value)
+
+            elif isinstance(value, datetime):
+                mapping[key] = str(value)
+        return mapping
+
+    async def run(self, payload: dict, in_edge=None) -> Result:
+
+        configuration = ElasticEmail.Configuration()
+        configuration.api_key['apikey'] = self.credentials.api_key
+
+        dot = self._get_dot_accessor(payload)
+        traverser = DictTraverser(dot)
+
+        email = dot[self.config.email]
+
+        mapping = traverser.reshape(self.config.additional_mapping)
+        mapping = self.parse_mapping(mapping)
+
+        contact_other = {}
+        contact_status = mapping.get('status')
+        if contact_status:
+            del mapping['status']
+
+        contact_other['status'] = ContactStatus(contact_status or "Active")
+        email_list_names = mapping.get('list_names')
+        if email_list_names:
+            email_list_names = email_list_names.split(',')
+            del mapping['list_names']
+
+        first_name = mapping.get('first_name')
+        if first_name:
+            del mapping['first_name']
+            contact_other['first_name'] = first_name
+
+        last_name = mapping.get('last_name')
+        if last_name:
+            del mapping['last_name']
+            contact_other['last_name'] = last_name
+
```

Comment:
if there is still fields they should be added to the miscellaneous area 

```suggestion
       if mapping:
            contact_other['custom_fields'] = mapping,
```